src/btree/bt_split.c

```diff
@@ -1431,20 +1431,31 @@ __split_multi_inmem(WT_SESSION_IMPL *session, WT_PAGE *orig, WT_MULTI *multi, WT
          * size of the page is correct.
          */
         if (supd->onpage_upd != NULL && !F_ISSET(S2C(session), WT_CONN_IN_MEMORY)) {
+            tmp = supd->onpage_tombstone != NULL ? supd->onpage_tombstone : supd->onpage_upd;
```

Comment:
Add a comment here:
"If there is an on-page tombstone we need to remove it as well while performing update restore eviction."

Reply:
Thanks, fixed in 9169207